src/english_score_tuning.py

```diff
@@ -0,0 +1,204 @@
+import click
+import pandas as pd
+import pickle
+import sys
+from sklearn.compose import make_column_transformer
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import (
+    OneHotEncoder,
+    OrdinalEncoder,
+    StandardScaler,
+    FunctionTransformer,
+)
+from sklearn.linear_model import Ridge, Lasso
+from sklearn.model_selection import RandomizedSearchCV
+from scipy.stats import loguniform
+from sklearn.metrics import PredictionErrorDisplay
+from sklearn.model_selection import train_test_split
+from search_top_models import fit_and_return_top_models
+from show_feat_coeff import show_feat_coeff
+
+SCORING = {"RMSE": "neg_root_mean_squared_error", "R squared": "r2"}
+
+
+@click.command()
+@click.option("--verbose", "-v", is_flag=True, help="Will print verbose messages.")
+@click.option(
+    "--train",
+    default="data/raw/train_dataset.csv",
+    help="Path to the training data.",
+)
+@click.option(
+    "--test",
+    default="data/raw/test_dataset.csv",
+    help="Path to the test data.",
+)
+@click.option(
+    "--output_dir",
+    default="results/models/",
+    help="Output directory, default is `results/models/`",
+)
+@click.option(
+    "--preprocessor_path",
+    default="results/models/preprocessor/preprocessor.pkl",
+    help="Path to the preprocessor object (default: results/models/preprocessor/preprocessor.pkl)",
+)
```

Comment:
Can we add a click option to store the fit preprocessor (so that I can access it in my new script):

@click.option(
    "--fit-pickle-to",
    type=str,
    help="Path to where the preprocessor (fitted to the training data) should be written ex: results/models/fit_preprocessor",
)